### PR TITLE
DOC:random:Reorganize front page

### PR DESCRIPTION
**Pull request by ilayn**
Relevant recent discussion and context https://github.com/scipy/scipy/issues/14322

Related : #20121 

Disclosure: I don't know much about the random internals but tried to clean up the text by putting what I want to see as an outsider upon landing on this page. I didn't want to overdo it hence it is quite limited in scope. But happy to expand it if need be.

Please feel free to add your own commits on top of this, would be appreciated. And fix my english along the way maybe?

### CODE REVIEW

File: doc/source/reference/random/index.rst [hunk 1]
```diff
@@ -18,41 +85,36 @@ to use those sequences to sample from different statistical distributions:
...
-methods to obtain samples from different distributions.  By default,
-`Generator` uses bits provided by `PCG64` which has better statistical
-properties than the legacy `MT19937` used in `RandomState`.
+:ref:`legacy` for the complete list. At this point, we have established
+the `Generator` based way-of-working still the legacy API is available.
+While this might seem like, we offer two options, it is quite the
+contrary. We sincerely hope that we minimized the annoyance during
+these changes, however we also hope that users will eventually stop
+using the legacy API which would allow us deprecate and remove the
+legacy API.
```
Comment (MatteoRaso) [inline doc/source/reference/random/index.rst:108]:
```suggestion
:ref:`legacy` for the complete list. While the legacy API is still available,
it will eventually be deprecated in the future, so we strongly advise using
the new API instead of the legacy API.
```
The paragraph was hard to understand, so I replaced it.
```

File: doc/source/reference/random/index.rst [hunk 2]
```diff
@@ -7,6 +7,73 @@
...
+Quick Reference
+---------------
+
+If you want to draw samples from various distributions, below is probably what you
+see in the old docs and elsewhere, roughly speaking, import ``random`` module and then
+use functions from the :ref:`legacy` section, such as
+
+.. code-block:: python
+
+  # pre-NumPy 1.17, nostalgic style,
```
Comment (MatteoRaso) [inline doc/source/reference/random/index.rst:28]:
```suggestion
  # pre-NumPy 1.17
```
This is just a bit of a nitpick, but I don't think "nostalgic style" should be there.

**Review by MatteoRaso — CHANGES_REQUESTED** [review]
Thanks for the PR. I've made a few suggestions, mostly to fix grammatical errors and missteps.

File: doc/source/reference/random/index.rst [hunk 3]
```diff
@@ -7,6 +7,73 @@
 Random sampling (:mod:`numpy.random`)
 =====================================
 
+NumPy facilitates various tools for generating pseudo-random numbers. This
+also includes providing means to create repeatable sequences for reproducible
+pseudo-random processes. As NumPy random module has undergone substantial changes
+first the quick reference for the major differences are laid out to demonstrate
+what the usage style used to be pre NumPy v1.17 and what currently is for
+the impatient reader.
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:15]: Not that I care too much about this but impatient reader is a very common phrase.  image

File: doc/source/reference/random/index.rst [hunk 4]
```diff
@@ -7,6 +7,73 @@
...
+
+.. _random-quick-start:
+
+Quick Reference
+---------------
+
+If you want to draw samples from various distributions, below is probably what you
+see in the old docs and elsewhere, roughly speaking, import ``random`` module and then
+use functions from the :ref:`legacy` section, such as
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:24]: Official docs don't mean we can predict where the user might encounter this. It's not a definitive source so academic certainty is not a prerequisite for the style here.

File: doc/source/reference/random/index.rst [hunk 5]
```diff
@@ -7,6 +7,73 @@
...
+   known as a "seed", such that the randomized procedures in the existing
+   scope can be reproduced. While this worked in many scenarios, the seed
+   setting caused some nontrivial challenges such as leaking to other processes
+   that suppoesed to use their own random processes or affecting other packages
+   that relied on NumPy and many other interesting bugs.
+
+   This situation changed when NumPy started offering a rather finer control
+   to initialize and contain these processes. As you can see from the last
+   example, you don't need to limit the scope of the seeding or implement more
+   hacks to contain the effects of the (deprecated) `seed`.
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:69]: Yes but I think we have quite orthogonal styles, you are putting way too much unnecessary details on the front page about the Generators themselves. A seed example is given already in the quick reference part. So the idea is to stop users using `np.random.seed` not to teach them the shiny Generators. 

In other words, the correctness is not the first objective here, providing affinity is. The gory details will come later anyways.

File: doc/source/reference/random/index.rst [hunk 6]
```diff
@@ -7,6 +7,73 @@
...
+   
+   Moreover, the current infrastructure takes a different approach to producing
+   random numbers from the legacy `RandomState` object.
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:72]: Ah, this is from the original page. I don't know much about the details to be honest. Just reordered it.

File: doc/source/reference/random/index.rst [hunk 7]
```diff
@@ -18,41 +85,36 @@ to use those sequences to sample from different statistical distributions:
...
-distributions. See `NEP 19 <https://www.numpy.org/neps/
-nep-0019-rng-policy.html>`_ for context on the updated random Numpy number
-routines. The legacy `RandomState` random number routines are still
-available, but limited to a single BitGenerator. See :ref:`new-or-different` 
+In short, BitGenerators are the parts responsible for the randomization
+and the state, whereas Generators are responsible for utilizing these
+randomized inputs.
+
+The Generators can be initialized with a number of different BitGenerators.
+They expose many different probability distributions. For the context
```
[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:93]: Haha, wait my english is not that bad. This is clearly a stylistic choice. Let's not confuse style with grammar.

File: doc/source/reference/random/index.rst (hunk 1 above)
Comment (ilayn) [reply doc/source/reference/random/index.rst:108]: what was hard to understand?

File: doc/source/reference/random/index.rst (hunk 2 above)
Comment (ilayn) [reply doc/source/reference/random/index.rst:28]: I agree. Will delete it.

File: doc/source/reference/random/index.rst [hunk 8]
```diff
@@ -18,41 +85,36 @@ to use those sequences to sample from different statistical distributions:
...
 
-Since Numpy version 1.17.0 the Generator can be initialized with a
-number of different BitGenerators. It exposes many different probability
-distributions. See `NEP 19 <https://www.numpy.org/neps/
-nep-0019-rng-policy.html>`_ for context on the updated random Numpy number
-routines. The legacy `RandomState` random number routines are still
-available, but limited to a single BitGenerator. See :ref:`new-or-different` 
+In short, BitGenerators are the parts responsible for the randomization
+and the state, whereas Generators are responsible for utilizing these
+randomized inputs.
```
[the remark this answers is not in the document]
Comment (h-vetinari) [reply doc/source/reference/random/index.rst:90]: I disagree, I think it's a useful summary, as the previous explanation is more precise but less accessible.

File: doc/source/reference/random/index.rst (hunk 1 above)
Comment (h-vetinari) [reply doc/source/reference/random/index.rst:108]: > what was hard to understand?

At least the first sentence is hard to parse ("we have established [x] still the ..."). I also don't think we should try to "sincerely hope" and frame a fact like its opposite - we _are_ offering two options, but we want (not hope) to transition from one to the other. If Matteo hadn't already posted his suggestion, I would have proposed something very similar.

File: doc/source/reference/random/index.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:93]: >Haha, wait my english is not that bad. This is clearly a stylistic choice. Let's not confuse style with grammar.

Your paragraph is grammatically correct, but English has right and wrong ways of styling. The way you styled these sentences is only done when writing something for somebody who you know struggles with reading English (e.g. writing books for small children). Like I said, it's a very confusing language, but this styling choice sticks out a lot for people who know English as a first language.

File: doc/source/reference/random/index.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:15]: >Not that I care too much about this but impatient reader is a very common phrase.

Yes, but those are more informal contexts. In any case, the main problem with that paragraph isn't the use of the phrase "impatient reader", it's the grammatical errors.

File: doc/source/reference/random/index.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:24]: >Official docs don't mean we can predict where the user might encounter this.

Yes, I left the "and elsewhere" there. When I wrote "would see in the old docs", it's implied that I meant that the reader would see it if they looked through the old docs, not that they actually saw it there.

File: doc/source/reference/random/index.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:90]: >I disagree, I think it's a useful summary, as the previous explanation is more precise but less accessible.

Good point. I looked at the file after it was formatted, and it flowed a lot better than I thought it would when I was looking at the unformatted text. @ilayn, disregard this suggestion.

File: doc/source/reference/random/index.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:69]: Actually, I took a look at the official documentation, and I see where the confusion is coming from. You're talking about `np.random.seed`, but when I was reading your changes, I was thinking about the `seed` parameter. I've changed my suggestion a bit to make that clearer.

**Comment by rkern** [comment]
Thank you @ilayn, I appreciate the work and deliberation you have put into this. I think this ultimately still doesn't quite go in the direction needed to solve what I think are the two main problems with the text. I have an alternative PR at gh-23585 with my full reasoning in the PR description. It was very helpful to have this PR and discussion to sharpen my thoughts about what this page needs.

**Comment by ilayn** [comment]
Ah missed your message. Thanks for the heads up and the work

**Comment by rgommers** [comment]
Thanks both of you for diving in and working on fixing this pain point!

File: doc/source/reference/random/index.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/source/reference/random/index.rst:90]: Hey @MatteoRaso, while I appreciate you jumping in and helping to review PRs, your tone and review style isn't really okay I'm afraid. There are a couple of issues I see:

1. Something like _"`<username>`, disregard this suggestion."_ sounds like an order (which is bad). You could have inserted the word "please" instead, or just left it out completely and hit the "Resolve conversation" button.
2. Starting reviewing with detailed grammatical points and suggested rephrases is an anti-pattern. Please try to focus on the big picture first. In case of this PR, that's how the information is structured, and whether the examples and points made are clear/helpful.

I've noticed this on a few other PRs as well. That includes your own PRs - _"Any updates to this PR?"_ is not how to ask for attention from very busy reviewers, I suggest trying something like _"It's been 10 days and I believe I have addressed all previous comments, can you please have another look at this PR?"_. 

Can you please read https://numpy.org/devdocs/dev/reviewer_guidelines.html#communication-guidelines, try to phrase things in a more kind/empathetic fashion, and avoid nitpicking and grammar?

[the remark this answers is not in the document]
Comment (MatteoRaso) [reply doc/source/reference/random/index.rst:90]: @rgommers Sorry about that. I do try to be polite, but it's difficult for me to convey tone online. Yes, I'll be sure to read the reviewer guideline and carefully apply it in future reviews. @ilayn sorry if I came off as rude in this PR.

[the remark this answers is not in the document]
Comment (rgommers) [reply doc/source/reference/random/index.rst:90]: thanks, much appreciated!

[the remark this answers is not in the document]
Comment (ilayn) [reply doc/source/reference/random/index.rst:90]: I didn't quite take any offense but I can see how it can be taken the other way but regardless thanks all for the quick resolution. Hopefully, this will improve the overall docs for everyone.
